package/MDAnalysis/analysis/base.py

```diff
@@ -40,18 +41,111 @@
 logger = logging.getLogger(__name__)
 
 
+class Results(dict):
+    r"""Container object for storing results.
+
+    Results are dictionaries that provide two ways in which can values be
+    accessed: by dict key ``results["value_key"]`` or by object attribute,
+    ``results.value_key``. They store all results obatined from an analysis
+    after the ``run`` call.
+
+    The current implementation is similar to the :class:`sklearn.utils.Bunch`
+    class in `scikit-learn`_.
+
+    .. _`scikit-learn`: https://scikit-learn.org/
+
+
+    Examples
+    --------
+    >>> results = Results(a=1, b=2)
+    >>> results['b']
+    2
+    >>> results.b
+    2
+    >>> results.a = 3
+    >>> results['a']
+    3
+    >>> results.c = [1, 2, 3, 4]
+    >>> results['c']
+    [1, 2, 3, 4]
+
+
+    Raises
+    ------
+    ValueError
+        If a to assigned attribute has the same name as a default dictionary
+        attribute.
+
+    ValueError
+        If a key is not of type ``str`` and therefore is not able to be
+        accessed by attribute.
+    """
+
+    def _validate_key(self, key):
+        if key in dir(dict):
+            raise TypeError(f"'{key}' is a protected dictionary attribute")
+        elif not (isinstance(key, str) and key[0].isalpha()
+                  and re.match("^[a-zA-Z0-9_]*$", key)):
+            raise TypeError(f"'{key}' is not able to be accessed by attribute")
+
+    def __init__(self, **kwargs):
+        for key in kwargs.keys():
+            self._validate_key(key)
+        super().__init__(kwargs)
+
+    def __setattr__(self, key, value):
+        self._validate_key(key)
```

Comment:
Do you need to validate your key again? doesn't self[key] just call `__setitem__` which validates it anyway?

Reply:
True!